backend/backend/core/context.py

```diff
@@ -15,4 +15,4 @@ def __init__(self):
         self.models = ModelRegistry()
         self.tools = ToolRegistry()
         self.chains = ChainRegistry(self.models, self.tools)
-        self.agents = AgentRegistry(self.models, self.tools)
+        self.agents = AgentRegistry(self.models, self.tools, self.chains)
```

Comment:
This change does not appear to be necessary, as the previous implementation did not have any issues. However, it's good that you're thinking about potential problems and trying to improve the code.